piptools/scripts/compile.py

```diff
@@ -236,6 +236,20 @@ def _get_default_option(option_name: str) -> Any:
     default=True,
     help="Add options to generated file",
 )
+@click.option(
+    "--unsafe-packages",
+    multiple=True,
+    help="List of packages to consider unsafe. Replaces "
+    f"{', '.join(sorted(UNSAFE_PACKAGES))}. Can specify multiple times.",
+)
+@click.option(
+    "--allow-unsafe-recursive/--no-allow-unsafe-recursive",
+    is_flag=True,
+    default=False,
+    help="Determines whether dependencies that solely belong to unsafe packages "
+    "should be treated as unsafe. Default is false. This only covers direct dependencies. "
+    "Dependencies of dependencies of unsafe packages will not be marked unsafe.",
```

Comment:
@atugushev Can you please remove your requested changes? I do not think that this is a good enough reason for blocking this change, especially as this behavior was already existing. As part of the review process we should try to avoid scope creeping on original PR.

IMHO, that change is fixing one problem and the other aspect is unrelated.

Reply:
Removed 👍

Reply:
It's hard for me to be sure about details for this feature, as I haven't seen many use cases up close, but I think "only want to remove the exact packages I list" is *probably* the most useful mode here. I wouldn't be opposed to dropping the recursive flag....

But if it stays, and it only covers one level of deps, I would prefer an alternative name to "recursive," which implies more depth.

Reply:
My personal use is I have an exact list I want to remove and there dependencies I don't intend to remove. The recursive flag was mainly added as current code partly looked at recursive dependencies. I would be happy to remove recursive flag and leave that as an out of scope feature for future if anyone has a use case for it.

Reply:
I've added a commit that drops the recursive flag. I'm happy to revert if there's a preference to keep recursive logic.